api/src/main/java/org/eclipse/microprofile/concurrent/spi/ThreadContextProvider.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrent.spi;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * <p>Third party providers of thread context implement this interface to enable the
+ * provided type of context to participate in thread context capture & propagation
+ * when the MicroProfile Concurrency specification's <code>ManagedExecutor</code>
+ * and <code>ThreadContext</code> are used to create and contextualize dependent
+ * actions and tasks.</p>
+ *
+ * <p>Application code must never access the classes within this <code>spi</code>
+ * package. Instead, application code uses the <code>ManagedExecutor</code> and
+ * <code>ThreadContext</code> interfaces.</p>
+ *
+ * <p>The <code>ThreadContextProvider</code> implementation and related classes are
+ * packaged within the third party provider's JAR file. The implementation is made
+ * discoverable via the standard <code>ServiceLoader</code> mechanism. The JAR file
+ * that packages it must include a file of the following name and location,</p>
+ *
+ * <code>META-INF/services/org.eclipse.microprofile.concurrent.spi.ThreadContextProvider</code>
+ *
+ * <p>The content of the aforementioned file must be a single line that specifies the
```

Comment:
Actually it may be more than one line, each with a different implementation of the interface.

Reply:
Thanks - good catch.  We definitely want to allow for multiple.  I have added a commit correcting the language here to indicate at least one instead of single.